prometheus/registry.hh

```diff
@@ -23,10 +21,12 @@ namespace prometheus {
       CollectorRegistry();
       ~CollectorRegistry();
 
+      typedef Collector::collection_type collection_type;
```

Comment:
Is there a specific reason you're re-typedefing here instead of using `using` statements?